pkg/manifestcomparators/comp_conditions_must_have_proper_ssa_tags.go

```diff
@@ -0,0 +1,234 @@
+package manifestcomparators
+
+import (
+	"fmt"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+type conditionsMustHaveProperSSATags struct{}
+
+func ConditionsMustHaveProperSSATags() CRDComparator {
+	return conditionsMustHaveProperSSATags{}
+}
+
+func (conditionsMustHaveProperSSATags) Name() string {
+	return "ConditionsMustHaveProperSSATags"
+}
+
+func (conditionsMustHaveProperSSATags) WhyItMatters() string {
+	return "Conditions should follow the standard schema included in  " +
+		"https://github.com/kubernetes/apimachinery/blob/release-1.29/pkg/apis/meta/v1/types.go#L1482-L1542" +
+		"and collection of conditions should be treated as a map with a key of type. " +
+		"This is indicated in kubebuilder tags with '// +listType=map' and '// +listMapKey=type'."
+
+}
+
+func (c conditionsMustHaveProperSSATags) Validate(crd *apiextensionsv1.CustomResourceDefinition) (ComparisonResults, error) {
+	errsToReport := []string{}
+
+	for _, newVersion := range crd.Spec.Versions {
+		conditionsWithoutMapListType := []string{}
+		conditionsWithoutListMapKeysType := []string{}
+		invalidConditionsProperties := []string{}
+		SchemaHas(newVersion.Schema.OpenAPIV3Schema, field.NewPath("^"), field.NewPath("^"), nil,
+			func(s *apiextensionsv1.JSONSchemaProps, fldPath, simpleLocation *field.Path, _ []*apiextensionsv1.JSONSchemaProps) bool {
+				if s.Type != "array" {
+					return false
+				}
+
+				if !strings.Contains(simpleLocation.String(), ".conditions") {
+					return false
+				}
+
+				errs := validateConditionProperties(s.Items.Schema.Properties)
+				if len(errs) != 0 {
+					invalidConditionsProperties = append(invalidConditionsProperties, errs...)
+				}
+
+				if s.XListType == nil || *s.XListType != "map" {
+					conditionsWithoutMapListType = append(conditionsWithoutMapListType, simpleLocation.String())
+				}
+
+				if len(s.XListMapKeys) == 0 || !stringSliceContainsString(s.XListMapKeys, "type") {
```

Comment:
We want it to be exactly `type` so that clients get the behavior they expect without extraneous keys.